### ISSUE TITLE
Cant find "DedicatedServer" anywhere

### ISSUE THREAD

**Issue by SuperNov4DZN** [escapingnetwork/core-keeper-dedicated#50]
Hi, I used docker-compose to create a server using the image provided here and after some trouble with permissions (running with wrong user) I was able to get the server up and running, I can connect fine and play fine.. the only problem is there I cant find the "config folder" anywhere, it looks like the server isnt even creating it, the folders that I have right now (inside of server-files) are the following:

![image](https://github.com/user-attachments/assets/2eec4e9a-79be-4511-bfb0-4264da3ca37d)

"server-data" has nothing in it.

can anyone help me with that?

**Comment by SuperNov4DZN** [comment]
This is my core.env file right now:

```
WORLD_INDEX=0
WORLD_NAME=Core Keeper Server
WORLD_SEED=0
WORLD_MODE=0
GAME_ID=(I've set the game id to something 29 chars long)
DATA_PATH=
MAX_PLAYERS=10
DISCORD=1
DISCORD_HOOK=(myhook was here)
SEASON=-1
SERVER_IP=
SERVER_PORT=
```

**Comment by Micke90s** [comment]
Hi, I just looked on my running container:

![image](https://github.com/user-attachments/assets/5714f4a8-41b4-4748-bac8-0c80049d2036)

There are the two main folders `core-keeper-data` and `core-keeper-dedicated`

* `core-keeper-data` \
This directory contains the configuration and the save games 
* `core-keeper-dedicated` \
This directory contains the application (seems to be your screenshot)

I think the file you are looking for should be `~/core-keeper-data/ ServerConfig.json`

**Comment by SuperNov4DZN** [comment]
Uhm, this is inside the container, right? But wasn't it supposed to be "accessible" via the folder "Server-Data" that I created "locally" ? My folder structure is:

`~/corekeeper_server/server-files`
`~/corekeeper_server/server-data`
`~/corekeeper_server/core.env`
`~/corekeeper_server/docker-compose.yml`

Inside of `docker-compose.yml` I've set the `volumes` to:

```
- ~/corekeeper_server/server-files:/home/steam/core-keeper-dedicated
- ~/corekeeper_server/server-data:/home/steam/core-keeper-data
````

By doing that I thought my server config files would go to my `server-data`, the `server-files` work just like that, have I done something wrong?

**Comment by SuperNov4DZN** [comment]
Another strange thing, just looked at what was inside my container and I have nothing under `core-keeper-data`

![Screenshot_20240906-094625_1.jpg](https://github.com/user-attachments/assets/8cf33b33-af3a-48b2-bcdf-23ae885e965c)

**Comment by Micke90s** [comment]
Could you please try to set the line `DATA_PATH=` from your `core.env` to `DATA_PATH=/home/steam/core-keeper-data`.

**Comment by Makerblaker** [comment]
I am facing the same issue. Anytime the container is restarted, all data is lost.

**Comment by SuperNov4DZN** [comment]
> Could you please try to set the line `DATA_PATH=` from your `core.env` to `DATA_PATH=/home/steam/core-keeper-data`.
> 
> Update:
> Should already be fixed with #28 - Fix may only work if environment variable is not present. 🤔 Could be the problem. 
> 
> Your save games should be available under ` /home/steam/.config/Pugstorm/Core Keeper/DedicatedServer` within the container.

Uhmmm, got it.. I'll make a backup of this folder (since I already played in the server for almost 2 hours), set the env path and try again, question, do I need to build the whole thing again running `docker-compose up` or just restarting the container is enough?

**Comment by Makerblaker** [comment]
I set the `DATA_PATH=` in the env file, and it seems to work now.
```shell
-batchmode -logfile CoreKeeperServerLog.txt -world 0 -worldname Purple Drop Server -worldseed 0 -worldmode 0 -datapath /home/steam/core-keeper-data -maxplayers 10 -season -1
```

**Comment by SuperNov4DZN** [comment]
> I set the `DATA_PATH=` in the env file, and it seems to work now.
> ```shell
> -batchmode -logfile CoreKeeperServerLog.txt -world 0 -worldname Purple Drop Server -worldseed 0 -worldmode 0 -datapath /home/steam/core-keeper-data -maxplayers 10 -season -1
> ``` 

Okay, so, no need to compose again? Just a simple restar is enough?

**Comment by Makerblaker** [comment]
> > I set the `DATA_PATH=` in the env file, and it seems to work now.
> > ```shell
> > -batchmode -logfile CoreKeeperServerLog.txt -world 0 -worldname Purple Drop Server -worldseed 0 -worldmode 0 -datapath /home/steam/core-keeper-data -maxplayers 10 -season -1
> > ```
> 
> Okay, so, no need to compose again? Just a simple restart is enough?

I don't think so. I would suggest connecting to the container and copying the `~/core-keeper-data/CoreKeeperServer_Data` so you don't lose your save.

**Comment by Micke90s** [comment]
@SuperNov4DZN I think that you have to use `docker-compose down` and `docker-compose up` to set the environment variables. This will cause that the container is rebuild. Please backup your save games to the host and rebuild it.

**Comment by SuperNov4DZN** [comment]
> @SuperNov4DZN I think that you have to use `docker-compose down` and `docker-compose up` to set the environment variables. This will cause that the container is rebuild. Please backup your save games to the host and rebuild it. 
> 
> 

Ok, thanks for the help! Appreciate it ♥️, I'll give it a try asap.

**Comment by arguser** [comment]
> Could you please try to set the line `DATA_PATH=` from your `core.env` to `DATA_PATH=/home/steam/core-keeper-data`.
> 
> Update: Should already be fixed with #28 - Fix may only work if environment variable is not present. 🤔 Could be the problem.
> 
> Your save games should be available under ` /home/steam/.config/Pugstorm/Core Keeper/DedicatedServer` within the container.

I just created a server without the `DATA_PATH` environment variable and worked as expected. So you are probably correct.

**Comment by SuperNov4DZN** [comment]
Okay, I did what @Micke90s told me to:

Changed `core.env` `DATA_PATH:/home/steam/core-keeper-data`
 
Then `docker-compose down` `docker-compose up -d`

Waited for a few seconds, got the notification on discord with the GameID, and after checking I now have everything on the right place, I have access to the server config files both on `host/server-data` and `container/core-keeper-data`

Worked like a charm, again, thanks for the help everyone, I'll be closing this issue.
